src/lib/blkalloc/varsize_blk_allocator.cpp

```diff
@@ -626,6 +626,24 @@ blk_count_t VarsizeBlkAllocator::alloc_blks_direct(blk_count_t nblks, blk_alloc_
     return (nblks - nblks_remain);
 }
 
+//since this function will only be called during HS recovery, we can safe to update the cache bitmap directly without
+//touching the slab caches.
+BlkAllocStatus VarsizeBlkAllocator::mark_blk_allocated(BlkId const& bid) {
+    BlkAllocPortion& portion = blknum_to_portion(bid.blk_num());
+    {
+        auto const start_blk_id = portion.get_portion_num() * get_blks_per_portion();
```

Comment:
Can we move line: 634 ~ 639 inside "ifndef NDEBUG" to speed up a little for release build since they are just doing some assert in debug mode?

Reply:
but on L636, we need to hold the  portion lock, right?

Reply:
right, 636 should be out of the macro.